src/Orleans/Runtime/GrainReference.cs

```diff
@@ -15,8 +14,6 @@ namespace Orleans.Runtime
     [Serializable]
     public class GrainReference : IAddressable, IEquatable<GrainReference>, ISerializable
     {
-        private static readonly Action<Message, TaskCompletionSource<object>> ResponseCallbackDelegate = ResponseCallback;
```

Comment:
I think the point of this was to avoid many allocations of the cast to Action. Perhaps removing the static keyword is good enough? Or would the allocations per GrainReference instance would be so few that this optimization doesn't make any more sense?